src/python/botan3.py

```diff
@@ -2123,6 +2143,59 @@ def clear_bit(self, bit):
     def set_bit(self, bit):
         _DLL.botan_mp_set_bit(self.__obj, c_size_t(bit))
 
+
+class OID:
+    def __init__(self, obj=None):
+        if not obj:
+            obj = c_void_p(0)
+        self.__obj = obj
```

Comment:
I intended to use the same `def __init__(self, obj=c_void_p(0)):` that `PublicKey` and `PrivateKey` use, but as it turns out python does _not_ create a new `c_void_p` everytime this is called, but instead creates it only once and then reuses it, leading to every instance of `OID` writing to the same memory. I would consider this unintended behavior, and thus solved it like so. `PrivateKey` and `PublicKey` get around this by meticulously creating new `c_void_p`s, and then passing that to their constructor. Should I adapt these to in the OID way, or leave them as is?

Reply:
Yes I ran into exactly this issue at some point and was quite surprised! Using `=None` as here is probably the way to go.